flink-queryable-state/flink-queryable-state-client-java/src/main/java/org/apache/flink/queryablestate/network/Client.java

```diff
@@ -133,7 +134,7 @@ public String getClientName() {
 	}
 
 	public CompletableFuture<RESP> sendRequest(final InetSocketAddress serverAddress, final REQ request) {
-		if (shutDown.get()) {
+		if (!clientShutdownFuture.compareAndSet(null, null)) {
```

Comment:
this should use ´clientShutdownFuture.get() != null` for clarity.

Reply:
Well, I agree that the code looks clearer, but the `compareAndSet` guarantees atomicity. While `get()` and then compare, does not.

Reply:
But the atomicity doesn't get you anything. Which case wouldn't be covered?

This check can only handle the case when sendRequest() is called after shutdown(), regardless of whether you use compareAndSet() or get(), so it shouldn't make a difference.